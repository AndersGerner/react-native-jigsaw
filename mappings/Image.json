{"name":"Image","tag":"Image","description":"A basic Image Component","category":"media","supports_list_render":false,"preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/Jigsaw/Image.png","layout":{"width":250,"height":250},"props":{"source":{"label":"Image Source","description":"The source of the image","editable":true,"required":true,"type":"localImage","value":null},"resizeMode":{"label":"Resize Mode","description":"Determines how to resize the image when the frame doesn't match the raw image dimensions","editable":true,"required":false,"value":"cover","type":"flatArray","options":["cover","contain","stretch","repeat","center"]}}}