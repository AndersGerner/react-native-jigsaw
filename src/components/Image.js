import { Image } from "react-native";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";

export default Image;

export const SEED_DATA = {
  name: "Image",
  tag: "Image",
  description: "A basic Image Component",
  type: COMPONENT_TYPES.primitive,
  supports_list_render: false,
  preview_image_url:
    "https://res.cloudinary.com/altos/image/upload/v1543957160/draftbit/Jigsaw/Image.svg",
  layout: {
    width: 200,
    height: 200
  },
  props: {
    source: {
      label: "Image Source",
      description: "The source of the image",
      editable: true,
      required: true,
      type: FORM_TYPES.localImage,
      value: null
    },
    resizeMode: {
      label: "Resize Mode",
      description:
        "Determines how to resize the image when the frame doesn't match the raw image dimensions",
      editable: true,
      required: false,
      value: null,
      type: FORM_TYPES.flatArray,
      options: ["cover", "contain", "stretch", "repeat", "center"]
    }
  }
};
