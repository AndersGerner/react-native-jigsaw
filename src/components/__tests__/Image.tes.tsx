import * as React from "react";
import renderer from "react-test-renderer";
import Image from "../Image";

it("renders image", () => {
  const tree = renderer
    .create(<Image source={require("../../assets/icon.png")} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
