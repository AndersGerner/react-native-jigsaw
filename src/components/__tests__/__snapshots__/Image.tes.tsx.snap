// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders image 1`] = `
<Image
  resizeMode="cover"
  source={1}
/>
`;
